internal/service/ec2/vpc_ipam_scope.go

```diff
@@ -109,18 +109,28 @@ func ResourceVPCIpamScopeRead(d *schema.ResourceData, meta interface{}) error {
 	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
 
 	scope, err := findIpamScopeById(conn, d.Id())
-	ipamId := strings.Split(*scope.IpamArn, "/")[1]
 
-	if err != nil && !tfawserr.ErrCodeEquals(err, InvalidIpamScopeIdNotFound) {
-		return err
+	if !d.IsNewResource() && tfresource.NotFound(err) {
+		log.Printf("[WARN] IPAM Scope (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error reading IPAM Scope (%s): %w", d.Id(), err)
 	}
 
-	if !d.IsNewResource() && scope == nil {
+	if scope == nil {
```

Comment:
`scope` should now never be `nil` after the changes to `findIpamScopeById` so I think this whole `if` block can be removed.

Reply:
Also noticed the `status` method can use the `NotFoundError` handling since the previous check is handled now within the `find` method